### ISSUE TITLE
iOS Static Code Analyzer crashes if .swift directories are present

### ISSUE THREAD

**Issue by 7a** [MobSF/Mobile-Security-Framework-MobSF#1533]
## ENVIRONMENT
**NOTE: Using the lastest Mobile Security Framework v3.1.4 Beta via Docker image** 
```
OS and Version: Linux-5.4.0-kali4-amd64-x86_64-with-glibc2.29, Dist: ubuntu 20.04 focal
Python Version: Not sure, but using docker, so whatever you have there
MobSF Version: Mobile Security Framework v3.1.4 Beta
```

## EXPLANATION OF THE ISSUE

iOS Static Code Analyzer crashes if any **directory** is named something.swift, seems a bug in how the filesystem is walked.

## STEPS TO REPRODUCE THE ISSUE

```
Uploaded a correctly formatted zip file for iOS Static analysis, see below for more details
```

## LOG FILE

First try:
```
[INFO] 09/Sep/2020 07:12:06 - Starting iOS Source Code and PLIST Analysis
[ERROR] 09/Sep/2020 07:13:48 - iOS Source Code Analysis
Traceback (most recent call last):
  File "/root/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/ios/code_analysis.py", line 72, in ios_source_analysis
    pfile.read_text('utf-8', 'ignore'), relative_java_path)
  File "/usr/lib/python3.8/pathlib.py", line 1227, in read_text
    with self.open(mode='r', encoding=encoding, errors=errors) as f:
  File "/usr/lib/python3.8/pathlib.py", line 1213, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
IsADirectoryError: [Errno 21] Is a directory: '/root/.MobSF/uploads/b6dbd25531b8e5729d6788f0eb4e441f/Pods/R.swift'
[INFO] 09/Sep/2020 07:13:48 - Fetching icon path
[ERROR] 09/Sep/2020 07:13:49 - Error Perfroming Static Analysis
Traceback (most recent call last):
  File "/root/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/ios/static_analyzer.py", line 199, in static_analyzer_ios
    list(set(code_analysis_dic['urls_list'])))
TypeError: 'NoneType' object is not subscriptable
[ERROR] 09/Sep/2020 07:13:49 - 'NoneType' object is not subscriptable
[ERROR] 09/Sep/2020 07:13:49 - Internal Server Error: /StaticAnalyzer_iOS/
```

Removed Pods.swift, using this command:

```
rm -rf R.swift
```

2nd try:

```
[ERROR] 09/Sep/2020 07:39:54 - iOS Source Code Analysis
Traceback (most recent call last):
  File "/root/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/ios/code_analysis.py", line 72, in ios_source_analysis
    pfile.read_text('utf-8', 'ignore'), relative_java_path)
  File "/usr/lib/python3.8/pathlib.py", line 1227, in read_text
    with self.open(mode='r', encoding=encoding, errors=errors) as f:
  File "/usr/lib/python3.8/pathlib.py", line 1213, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
IsADirectoryError: [Errno 21] Is a directory: '/root/.MobSF/uploads/56f60100212d67b53cea437f85983d8e/Pods/Target Support Files/R.swift'
[INFO] 09/Sep/2020 07:39:54 - Fetching icon path
[ERROR] 09/Sep/2020 07:39:55 - Error Perfroming Static Analysis
Traceback (most recent call last):
  File "/root/Mobile-Security-Framework-MobSF/StaticAnalyzer/views/ios/static_analyzer.py", line 199, in static_analyzer_ios
    list(set(code_analysis_dic['urls_list'])))
TypeError: 'NoneType' object is not subscriptable
[ERROR] 09/Sep/2020 07:39:55 - 'NoneType' object is not subscriptable
[ERROR] 09/Sep/2020 07:39:55 - Internal Server Error: /StaticAnalyzer_iOS/
```

ran this command before zipping again:

```
rm -rf "./Pods/Target Support Files/R.swift"
```

After doing that, the static analysis worked.

I would recommend to make the static analysis code smarter so .swift directories are not processed as files, then things will work more smoothly.

**Comment by x3n0nth3k** [comment]
This is already fixed in #1532

**Comment by ajinabraham** [comment]
This should be fixed by @x3n0nth3k in latest master yesterday. Also the image corresponding to the fix was auto build with `latest` tag and is available on DockerHub.
`sha256:7a80a6560c117880a2990c4ddd3524dd89805c8fd5e635e6c54a792781ec7828`

Verified this in the latest docker image

```bash
[INFO] 09/Sep/2020 19:42:22 - Mobile Security Framework v3.1.4 Beta
REST API Key: xxxx
[INFO] 09/Sep/2020 19:42:22 - OS: Linux
[INFO] 09/Sep/2020 19:42:22 - Platform: Linux-4.4.0-186-generic-x86_64-with-glibc2.29
[INFO] 09/Sep/2020 19:42:22 - Dist: ubuntu 20.04 focal
[INFO] 09/Sep/2020 19:42:22 - MobSF Basic Environment Check
[INFO] 09/Sep/2020 19:42:22 - Checking for Update.
[INFO] 09/Sep/2020 19:42:22 - No updates available.
[INFO] 09/Sep/2020 19:43:27 - MIME Type: application/zip FILE: archive.zip
[INFO] 09/Sep/2020 19:43:27 - Performing Static Analysis of Android/iOS Source Code
[INFO] 09/Sep/2020 19:43:27 - Starting Analysis on : Archive.zip
[INFO] 09/Sep/2020 19:43:27 - Extracting ZIP
[INFO] 09/Sep/2020 19:43:27 - Unzipping
[INFO] 09/Sep/2020 19:43:29 - Checking for ZIP Validity and Mode
[INFO] 09/Sep/2020 19:43:29 - Redirecting to iOS Source Code Analyzer
[INFO] 09/Sep/2020 19:43:29 - iOS Static Analysis Started
[INFO] 09/Sep/2020 19:43:29 - iOS Source Code Analysis Started
[INFO] 09/Sep/2020 19:43:29 - ZIP Already Extracted
[INFO] 09/Sep/2020 19:43:29 - Generating Hashes
[INFO] 09/Sep/2020 19:43:30 - Get Files, BIN Plist -> XML, and Normalize
[INFO] 09/Sep/2020 19:43:30 - iOS Info.plist Analysis Started
[INFO] 09/Sep/2020 19:43:30 - Finding Info.plist in iOS Source
[INFO] 09/Sep/2020 19:43:30 - Checking Permissions
[INFO] 09/Sep/2020 19:43:30 - Checking for Insecure Connections
[INFO] 09/Sep/2020 19:43:30 - Fetching Details from App Store: $(PRODUCT_BUNDLE_IDENTIFIER)
[WARNING] 09/Sep/2020 19:43:30 - Unable to get app details.
[INFO] 09/Sep/2020 19:43:30 - Starting iOS Source Code and PLIST Analysis
[INFO] 09/Sep/2020 19:43:51 - Performing Malware Check on extracted Domains
[INFO] 09/Sep/2020 19:43:51 - Updating Malware Database....
[INFO] 09/Sep/2020 19:43:53 - Finished Code Analysis, Email and URL Extraction
[INFO] 09/Sep/2020 19:43:53 - Fetching icon path
[INFO] 09/Sep/2020 19:43:53 - Detecting Firebase URL(s)
[INFO] 09/Sep/2020 19:43:53 - Connecting to DB
[INFO] 09/Sep/2020 19:43:53 - Saving to Database
```

<img width="791" alt="Screenshot 2020-09-09 at 3 45 28 PM" src="https://user-images.githubusercontent.com/4301109/92645897-93be2000-f2b3-11ea-975d-559546bf70d2.png">

Please reopen if you can still reproduce this on latest image.
